examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,366 @@
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="campaignId">ID of the campaign to which sitelinks are added.</param>
+        /// <param name="adGroupId">Optional ID of the ad group to which sitelinks are added.</param>
+        public void Run(GoogleAdsClient client, long customerId, long campaignId, long? adGroupId)
+        {
+            try
+            {
+                // Create a feed, which acts as a table to store data.  
+                // feed_data = create_feed(client, customer_id)
+                Hashtable feedData = CreateFeed(client, customerId);
+
+                // Create feed items, which fill out the feed table with data.
+                CreateFeedItems(client, customerId, feedData);
+
+                // Create a feed mapping, which tells Google Ads how to interpret this data to
+                // display additional sitelink information on ads.
+                CreateFeedMapping(client, customerId, feedData);
+                
+                // Create a campaign feed, which tells Google Ads which campaigns to use the
+                // provided data with.
+                
+                CreateCampaignFeed(client, customerId, campaignId, feedData);
+                // If an ad group is specified, limit targeting only to the given ad group.
+                if (adGroupId.HasValue)
+                {
+                    CreateAdGroupTargeting(client, customerId, feedData, adGroupId.Value);
+                }
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+                throw;
+            }
+        }
+
+        private Hashtable CreateFeed(GoogleAdsClient client, long customerId)
+        {
+            Feed feed = new Feed()
+            {
+                Name = $"Sitelinks Feed {DateTime.Now:yyyy-MM-dd hh:mm:ss}",
+                Origin = FeedOriginEnum.Types.FeedOrigin.User
+            };
+
+            // Specify the column name and data type. This is just raw data at this point,
+            // and not yet linked to any particular purpose. The names are used to help us
+            // remember what they are intended for later.
+            feed.Attributes.Add(CreateFeedAttribute("Link Text",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Link Final Url",
+                FeedAttributeType.UrlList));
+            feed.Attributes.Add(CreateFeedAttribute("Line 1",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Line 2",
+                FeedAttributeType.String));
+
+            FeedServiceClient feedServiceClient = client.GetService(Services.V3.FeedService);
```

Comment:
Nit: We generally do the service creation in the beginning of the function. Just a styling thing, nothing more.

Reply:
Gotcha.  Moved to be consistent.